internal/framework/status/setters.go

```diff
@@ -0,0 +1,215 @@
+package status
+
+import (
+	"slices"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+)
+
+// setter is a function that takes an object and sets the status on that object if the status has changed.
+// If the status has not changed, and the setter does not set the status, it returns false.
+type setter func(client.Object) bool
+
+func newNginxGatewayStatusSetter(clock Clock, status NginxGatewayStatus) func(client.Object) bool {
+	return func(object client.Object) bool {
+		ng := object.(*ngfAPI.NginxGateway)
+		conds := convertConditions(
+			status.Conditions,
+			status.ObservedGeneration,
+			clock.Now(),
+		)
+
+		if conditionsEqual(ng.Status.Conditions, conds) {
+			return false
+		}
+
+		ng.Status = ngfAPI.NginxGatewayStatus{
+			Conditions: conds,
+		}
+
+		return true
+	}
+}
+
+func newGatewayClassStatusSetter(clock Clock, gcs GatewayClassStatus) func(client.Object) bool {
+	return func(object client.Object) bool {
+		gc := object.(*v1beta1.GatewayClass)
+		status := prepareGatewayClassStatus(gcs, clock.Now())
+
+		if conditionsEqual(gc.Status.Conditions, status.Conditions) {
+			return false
+		}
+
+		gc.Status = status
+
+		return true
+	}
+}
+
+func newGatewayStatusSetter(podIP string, clock Clock, gs GatewayStatus) func(client.Object) bool {
+	return func(object client.Object) bool {
+		gw := object.(*v1beta1.Gateway)
+		status := prepareGatewayStatus(gs, podIP, clock.Now())
+
+		if gwStatusEqual(gw.Status, status) {
+			return false
+		}
+
+		gw.Status = status
+
+		return true
+	}
+}
+
+func newHTTPRouteStatusSetter(gatewayCtlrName string, clock Clock, rs HTTPRouteStatus) func(client.Object) bool {
+	return func(object client.Object) bool {
+		hr := object.(*v1beta1.HTTPRoute)
+		// statuses.GatewayStatus is never nil when len(statuses.HTTPRouteStatuses) > 0
```

Comment:
I wonder if this comment is still relevant

Reply:
Doesn't look like it. I'll remove.